devops/scripts/benchmarks/benches/base.py

```diff
@@ -99,13 +107,28 @@ def run_bench(
         ld_libraries = options.extra_ld_libraries.copy()
         ld_libraries.extend(ld_library)
 
-        result = run(
-            command=command,
-            env_vars=env_vars,
-            add_sycl=add_sycl,
-            cwd=options.benchmark_cwd,
-            ld_library=ld_libraries,
-        )
+        if unitrace_timestamp is not None:
```

Comment:
It's unintuitive that passing a unitrace timestamp triggers unitracing. Please, consider making a unitrace class and passing a `Unitrace` object or `None`. This would also allow encapsulating unitrace download and setup inside `Unitrace`'s constructor.